src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -359,46 +363,51 @@ private static string GetNextMethodName(ArrayBuilder<MethodSymbol> builder)
                         }
                     }
 
-                    // Hoisted method parameters (represented as locals in the EE).
-                    if (!_hoistedParameterNames.IsEmpty)
-                    {
-                        int localIndex = 0;
-                        foreach (var local in _localsForBinding)
-                        {
-                            // Since we are showing hoisted method parameters first, the parameters may appear out of order
-                            // in the Locals window if only some of the parameters are hoisted.  This is consistent with the
-                            // behavior of the old EE.
-                            if (_hoistedParameterNames.Contains(local.Name))
-                            {
-                                AppendLocalAndMethod(localBuilder, methodBuilder, local, container, localIndex, GetLocalResultFlags(local));
-                            }
-
-                            localIndex++;
-                        }
-                    }
-
-                    // Method parameters (except those that have been hoisted).
+                    var itemsAdded = new HashSet<string>();
                     int parameterIndex = m.IsStatic ? 0 : 1;
                     foreach (var parameter in m.Parameters)
                     {
                         var parameterName = parameter.Name;
-                        if (!_hoistedParameterNames.Contains(parameterName) &&
-                            GeneratedNames.GetKind(parameterName) == GeneratedNameKind.None &&
+                        if (GeneratedNames.GetKind(parameterName) == GeneratedNameKind.None &&
                             !IsDisplayClassParameter(parameter))
                         {
+                            itemsAdded.Add(parameterName);
                             AppendParameterAndMethod(localBuilder, methodBuilder, parameter, container, parameterIndex);
                         }
 
                         parameterIndex++;
                     }
 
+                    // iterator or async state machine
+                    if (!itemsAdded.Any())
```

Comment:
The comment seems misleading since `itemsAdded` will be empty if the source method has no parameters.

Reply:
Use `itemsAdded.Length == 0` to avoid allocating an `IEnumerator`.

Reply:
Thanks!

---
In reply to: [135911558](https://github.com/dotnet/roslyn/pull/21707#discussion_r135911558) [](ancestors = 135911558)